Provider/ImageProvider.php

```diff
@@ -72,12 +72,32 @@ public function getHelperProperties(MediaInterface $media, $format, $options = a
             $box = $this->resizer->getBox($media, $resizerFormat);
         }
 
+        $mediaWidth = $box->getWidth();
+
+        $srcset = array();
+        $context = $media->getContext();
+
+        foreach ($this->getFormats() as $providerFormat => $settings) {
+            // Check if format belongs to the current media's context
+            if (substr($providerFormat, 0, strlen($context)) == $context) {
```

Comment:
I stole that from there:
https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Thumbnail/FormatThumbnail.php#L79-L80